dtfs-central-api/api-tests/mocks/entities/azure-file-info-entity.ts

```diff
@@ -0,0 +1,12 @@
+import { AzureFileInfoEntity } from "@ukef/dtfs2-common";
+
+export const anAzureFileInfoEntity = (): AzureFileInfoEntity => {
```

Comment:
I've seen this kind of naming pattern before which tbh I quite like. It's a little different from the previous pattern of just calling stuff MOCK_AZURE... etc. although there didn't seem to be a pattern before that anyway and it was quite random. It's also nice this is a function than an object to use! Wouldn't mind moving to this

Reply:
I recognise this is quite different to current patterns, but yeah I really like using functions for getting test data so there is less possibility of dependency between tests and confusing test failures where one test modifies a mock object by accident and causes other tests to fail!

Reply:
I think the way we've normally done this is to treat `MOCK_<thing>` as a fully defined object for `<thing>` and then modify the fields that we want to for the specific test case, therefore removing the dependency on the shape of the mock object with respect to fields that we don't care about. For example, if we wanted to specifically test the `dateUploaded` property of a utilisation report, we would have a step like
```typescript
it('does something', () => {
  // Arrange
  const dateUploaded = new Date('2024-01-01');
  
  // by using MOCK_UTILISATION_REPORT, we remove any type errors without having to explicitly define all properties
  const reportWithUploadedDate: UtilisationReport = {
    ...MOCK_UTILISATION_REPORT,
    dateUploaded,
  };
  
  // Act
  ...
  
  // Assert
  expect(<someResult>).toEqual(expect.objectContaining({ dateUploaded }));
});
```
A pattern like this seems to just abstract away the "overwriting the fields we need and keeping everything else with valid but arbitrary values" part of the `reportWithUploadedDate` above, so I don't really think it's a different pattern so much as a more explicit way of doing what we've usually been doing. All that to say I don't think there are any risks in introducing this as a way of generating mock data for tests

Reply:
That being said, a safer pattern that is currently being added in FN-1859 is to have a class in `libs/common` called `UtilisationReportEntityMockBuilder` which creates and returns a `UtilisationReportEntity` with specified fields, defaulting to arbitrary values when those fields are net set in the mock creation. I think we should try and go with that pattern, but I also don't think we need to introduce that as part of this PR

Reply:
Yeah I think I just like this pattern above because it gets rid of the risk of someone going `ooh a mock, i'll use that directly` while doing a test that doesn't really need it too much and moving that spread pattern described essentially out to the function. 
In terms of a mock builder in libs/common rather than anywhere else, I think fitting those would work to be used inside these individual givenUploadedReport/aUtilisationReport, or are you thinking of also moving all helpers that say 'insert a utilisation report into the DB for an api-test' into libs/common as well and having a lot of test function helpers in there?

Worth noting we should be careful as to what we build as part of the libs/common package that gets built in all other services - right now we only build `./src` in the different apps and the api-tests etc. dont make it into deployed environments, and so if we put loads of testing logic in libs/common we'll need to do similar

Reply:
(fwiw i think this should be merged now independent of this, and we build on this approach in future PRs similar to what was just suggested, dont let this discussion block the merging :D )